Source/DataSources/CzmlDataSource.js

```diff
@@ -1801,17 +1813,15 @@ define([
         processPacketData(Number, polyline, 'granularity', polylineData.granularity, interval, sourceUri, entityCollection);
         processMaterialPacketData(polyline, 'material', polylineData.material, interval, sourceUri, entityCollection);
         processMaterialPacketData(polyline, 'depthFailMaterial', polylineData.depthFailMaterial, interval, sourceUri, entityCollection);
-        // Don't want to break CZML spec to keep this workaround even after followSurface has been deprecated from geometry.
-        // See https://github.com/AnalyticalGraphicsInc/cesium/pull/7582#discussion_r258695385
+
+        // for backwards compatibility, adapt CZML followSurface to arcType.
         if (defined(polylineData.followSurface) && !defined(polylineData.arcType)) {
-            if (polyline.followSurface) {
-                processPacketData(ArcType, polyline, 'arcType', ArcType.GEODESIC, interval, sourceUri, entityCollection);
-            } else {
-                processPacketData(ArcType, polyline, 'arcType', ArcType.NONE, interval, sourceUri, entityCollection);
-            }
-        } else {
-            processPacketData(ArcType, polyline, 'arcType', polylineData.arcType, interval, sourceUri, entityCollection);
+            var tempObj = {};
+            processPacketData(Boolean, tempObj, 'followSurface', polylineData.followSurface, interval, sourceUri, entityCollection);
+            polyline.arcType = createAdapterProperty(tempObj.followSurface, adaptFollowSurfaceToArcType);
         }
+
+        processPacketData(ArcType, polyline, 'arcType', polylineData.arcType, interval, sourceUri, entityCollection);
```

Comment:
Should this be in an `else`?  It's probably be simpler to do `if (defined(arcType)){...} else if (defined(followSurface){...}`

Reply:
Since both properties will be part of the CZML spec, a document could use both, in which case arcType overrides followSurface (since it's more powerful).  I rearranged the code a bit to maybe make that more clear.